Gym/DQN/main.swift

```diff
@@ -0,0 +1,190 @@
+// Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import PythonKit
+import TensorFlow
+
+// Initialize Python. This comment is a hook for internal use, do not remove.
+
+let np = Python.import("numpy")
+let gym = Python.import("gym")
+let plt = Python.import("matplotlib.pyplot")
+
+class TensorFlowEnvironmentWrapper {
+  let originalEnv: PythonObject
+
+  init(_ env: PythonObject) {
+    self.originalEnv = env
+  }
+
+  func reset() -> Tensor<Float> {
+    let state = self.originalEnv.reset()
+    return Tensor<Float>(numpy: np.array(state, dtype: np.float32))!
+  }
+
+  func step(_ action: Tensor<Int32>) -> (
+    state: Tensor<Float>, reward: Tensor<Float>, isDone: Tensor<Bool>, info: PythonObject
+  ) {
+    let (state, reward, isDone, info) = originalEnv.step(action.scalarized()).tuple4
+    let tfState = Tensor<Float>(numpy: np.array(state, dtype: np.float32))!
+    let tfReward = Tensor<Float>(numpy: np.array(reward, dtype: np.float32))!
+    let tfIsDone = Tensor<Bool>(numpy: np.array(isDone, dtype: np.bool))!
+    return (tfState, tfReward, tfIsDone, info)
+  }
+}
+
+func eval(agent: Agent) -> Float {
```

Comment:
Minor naming suggestions, based on the [Swift API Design Guidelines](https://swift.org/documentation/api-design-guidelines):

```suggestion
func evaluate(_ agent: Agent) -> Float {
```

Idiomatic Swift tends to [avoid abbreviations](https://swift.org/documentation/api-design-guidelines/#avoid-abbreviations) in favor of clarity. The `agent:` argument label can be omitted because it repeats type information.

Reply:
Didn't know that! Thank you for the link. I have made the changes.